packages/headless-styles/src/utils/helpers.ts

```diff
@@ -0,0 +1,41 @@
+import kebabCase from 'kebab-case'
+
+type StyleObject = Record<string, unknown>
+type NestedStyleValue = string | StyleObject
+
+function formatCSSPropName(propName: string) {
+  if (propName.includes('&')) {
+    return propName
+  }
+
+  return `${kebabCase(propName)}:`
+}
+
+function transformValue(style: NestedStyleValue): NestedStyleValue {
+  if (typeof style === 'string') {
+    return `${style};`.trim()
```

Comment:
If there's any whitespace at the end of `style`, it won't be trimmed (not sure if that matters). 
```suggestion
    return `${style.trim()};`
```